File: 002_reverse_words/reverse_words.py
def solution1(string):
    """
    Splits string into an array of words, reverses the order of words, 
    then creates a new string by joining the words together.
    """
    return ' '.join(string.split(' ')[::-1])

if __name__ == '__main__':
    test_string = 'How now brown cow'
    print(solution1(test_string))

File: 002_reverse_words/reverse_words.go
package main

import (
    "fmt"
    "strings"
)

func solution1(input string) string {
    words := strings.Split(input, " ")
    
    for i, j := 0, len(words) - 1; i < j; i, j = i+1, j-1 {
        words[i], words[j] = words[j], words[i]
    }

    return strings.Join(words, " ")
}

func main() {
    test_string := "How now brown cow"
    fmt.Println(solution1(test_string))
}
